### PR TITLE
Transfer remaining documentation from SourceForge

### LINKED ISSUE #2829: Transfer the remaining documentation from sourceforge
With the new RTD documentation, the old documentation material needs to be ported from [sourceforge](https://sourceforge.net/p/spinalcordtoolbox/wiki/Home/).

However, not _all_ the material needs to be ported:
- Some is old/obsolete and needs to be deleted
- Some will be ported in RTD
- Some will be incorporated in SCT course

### PR DESCRIPTION
**Pull request by joshuacwnewton**
### Related Issues/PRs

Fixes #2829.
Related to #2934. (Dependent on feedback)

### Description

The rendered documentation for branch `jn/2829-transfer-sourceforge-docs` can be found [here](https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/).

Task list:

 - [x] Image for `sct_propseg` (Result: [[link]](https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/user_section/command-line.html#sct-propseg))
 - [x] Image for `sct_dmri_separate_b0_and_dwi` (Result: [[link]](https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/user_section/command-line.html#sct-dmri-separate-b0-and-dwi))
 - [x] Missing references for `sct_propseg` (Result: [[link]](https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/user_section/command-line.html#sct-propseg))
 - [Template page](https://sourceforge.net/p/spinalcordtoolbox/wiki/MNI-Poly-AMU/)
     - [x] Write new guide for PAM50 template that is an analog to the old guide, including the following pages:
         - [x] [White matter atlas page](https://sourceforge.net/p/spinalcordtoolbox/wiki/White%20Matter%20atlas/)
         - [x] [Spinal levels page](https://sourceforge.net/p/spinalcordtoolbox/wiki/Spinal_levels/)
- [Tricks for `sct_register_multimodal`/`sct_register_to_template` page](https://sourceforge.net/p/spinalcordtoolbox/wiki/registration_tricks/)
     - [x] Ensure argparse parameter descriptions are up to date. ((Result: [[link]](https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/user_section/command-line.html#sct-register-multimodal))
     - [x] Move "Tips" section to script argparse description. (Result: [[link]](https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/user_section/command-line.html#sct-register-multimodal))
     - [x] Point `sct_register_to_template` to the `sct_register_multimodal` help description. (Result: [[link]](https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/user_section/command-line.html#sct-register-to-template))
- [Correcting `sct_propseg` page](https://sourceforge.net/p/spinalcordtoolbox/wiki/correction_PropSeg/)
     - [x] Copy over source material and fix formatting. (Result: [[link]](https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/user_section/tutorials/correcting_sct_propseg.html))

### CODE REVIEW

**Comment by joshuacwnewton** [comment]
I'm partially done with this pull request, but I'm facing some blockers:
* I don't yet understand the PAM50 template well enough to confidently write up-to-date guides on it. 
    * For example, the [Spinal levels](https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/overview/concepts.html#spinal-levels) information is out of date (there are more than 11 nifti files) but I'm not yet sure how to update this in a useful way.
* Some of the remaining changes involve the argparse descriptions, so they are blocked by #2842.

**Comment by joshuacwnewton** [comment]
I've now created a first draft of a PAM50 Template guide here: https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/overview/concepts/pam50.html

I do think the images could use some work.... the spinal levels image might be out of date. Also, there are no images for the template section (compare to the old guide: https://sourceforge.net/p/spinalcordtoolbox/wiki/MNI-Poly-AMU/)

But, I think this is good enough to invite feedback? I'll set this draft PR as ready for review.

**Comment by joshuacwnewton** [comment]
One other observation I had was, the current argparse helps for [`sct_register_multimodal`](https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/user_section/command-line.html#sct-register-multimodal) and [`sct_register_to_template`](https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/user_section/command-line.html#sct-register-to-template) are getting quite long and messy. I think that these would be good candidates for example/tutorial pages as part of #2857. 

I want to think about how best to organize such user guides. This PR introduces a ["Tutorials" section](https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/user_section/tutorials.html), but this is yet another section among many similarly-themed sections. (SCT Concepts/Command-Line Tools/Tutorials... they all provide "guides"). So, as a user, I wonder if it's clear where to look for information on a given topic.

**Comment by jcohenadad** [comment]
> One other observation I had was, the current argparse helps for [`sct_register_multimodal`](https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/user_section/command-line.html#sct-register-multimodal) and [`sct_register_to_template`](https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/user_section/command-line.html#sct-register-to-template) are getting quite long and messy. I think that these would be good candidates for example/tutorial pages as part of #2857.

👍 

> I want to think about how best to organize such user guides. This PR introduces a ["Tutorials" section](https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/user_section/tutorials.html), but this is yet another section among many similarly-themed sections. (SCT Concepts/Command-Line Tools/Tutorials... they all provide "guides"). So, as a user, I wonder if it's clear right now where to look for information on a given topic.

My advice would be to tailor our future doc refactoring around the datalad-type book idea. The earlier we start this project, the earlier we can think of the structure of the "final" doc. Of course, i am aware that all the restructuring we are doing now could (should) ultimately be ported to the book-like doc, but it might be easier to do think restructuring once we have a foundation, and an idea about how this book-like doc will display on a web browser (eg: appearance of the TOC, link between sections, incorporation of figures, etc.)

**Comment by joshuacwnewton** [comment]
> My advice would be to tailor our future doc refactoring around the datalad-type book idea.

I agree! To temper expectations slightly, though, the Dataland handbook is a massive effort coming from the output of [dozens of contributors](https://github.com/datalad-handbook/book/blob/master/README.md#contributors-). I can do my best to make gradual improvements, and to take cues from their design, but I feel a bit hesitant committing to all of the bells and whistles they have as I'm just one person. :sweat_smile: 

On reflection, I think it might be best to narrow the scope a bit, and focus on just a few key requirements to keep things a bit more realistic.

Perhaps we can continue this chat in #2857. I have a comment I want to make there. :)

**Comment by jcohenadad** [comment]
> I agree! To temper expectations slightly, though, the Dataland handbook is a massive effort coming from the output of dozens of contributors. I can do my best to make gradual improvements, and to take cues from their design, but I feel a bit hesitant committing to all of the bells and whistles they have as I'm just one person. 😅

you're absolutely right! to remove the pressure: what i envisioned was you setting up the foundation (backbone of the book, integration in RTD, etc.) and me being the one doing most of the content filling and curation.

**Review by jcohenadad — APPROVED** [review]

**Comment by joshuacwnewton** [comment]
@jcohenadad Ah, my apologies. Given the recent discussion about images and repository size, I realize I should not have committed the original images here. (Some of them are quite big.)

I will resize/compress them and address the commits in this PR so that they aren't preserved in the git history. Please don't merge yet. :)

**Comment by jcohenadad** [comment]
> I will resize/compress them and address the commits in this PR so that they aren't preserved in the git history. Please don't merge yet. :)

given that we use "squash and merge", these commits won't be preserved in the master branch AFAIK (which is one of the main reason why i like squash and merge)

File: documentation/source/overview/concepts/pam50.rst
Comment (jcohenadad) [inline documentation/source/overview/concepts/pam50.rst:82]:
```suggestion
* `neuropoly/template GitHub repository <https://github.com/neuropoly/template>`_
```

**Comment by jcohenadad** [comment]
@joshuacwnewton about figures and binary material for the figures: how about putting all these in a dedicated repos (eg: spinalcordtoolbox/doc-figures), so we won't be anxious about changing/resizing them in the future, as they won't be part of the main repos and won't add any fat to it.

**Comment by joshuacwnewton** [comment]
@jcohenadad I like this idea. 

I don't think I currently have the privileges to make a new repo under `spinalcordtoolbox`, though. Could you help with that? :)

**Comment by jcohenadad** [comment]
> @jcohenadad I like this idea.
> 
> I don't think I currently have the privileges to make a new repo under `spinalcordtoolbox`, though. Could you help with that? :)

created https://github.com/spinalcordtoolbox/doc-figures and added you as maintainer

File: documentation/source/overview/concepts/pam50.rst
Comment (joshuacwnewton) [inline documentation/source/overview/concepts/pam50.rst:37]: On closer look, I think the old title here is outdated. This is no longer just a "White matter atlas" now that GM regions have been added as well. 

Possibly this should be named "White and grey matter atlas"

**Comment by joshuacwnewton** [comment]
@jcohenadad I've updated the images to point to `spinalcordtoolbox/doc-figures`. I've also made a few other tweaks. (I updated some remaining references to SourceForge and I cleared up some unclear wording.)

Could you please re-review when you have a moment? (It may take a bit for the docs to refresh, so no rush.) Thanks in advance! :slightly_smiling_face: 

https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/index.html

As soon as this gets merged I can start on https://github.com/neuropoly/spinalcordtoolbox/issues/2934#issuecomment-717338928 as well, since it relies on the PAM50 page being finalized.

File: documentation/source/overview/concepts/pam50.rst
Comment (jcohenadad) [reply documentation/source/overview/concepts/pam50.rst:36]: EXCELLENT catch! yes, you're absolutely right-- i wanted to add a comment about that yesterday and then got caught into urgent stuff... "White and grey matter atlas" is great 👍

**Comment by jcohenadad** [comment]
It looks FANTASTIC! Since we're at it, I am preparing an image for the PAM50 template (like an overview)

**Comment by jcohenadad** [comment]
@joshuacwnewton could you please add this overview figure?
https://github.com/spinalcordtoolbox/doc-figures/tree/master/pam50

File: documentation/source/overview/concepts/pam50.rst
Comment (jcohenadad) [inline documentation/source/overview/concepts/pam50.rst:20]:
```suggestion
* It is compatible with the ICBM152 brain template (a.k.a. MNI template), allowing to conduct simultaneous brain/spine studies within the same referential system.
```

**Comment by joshuacwnewton** [comment]
> @joshuacwnewton could you please add this overview figure?
> https://github.com/spinalcordtoolbox/doc-figures/blob/master/pam50/pam50_template.png

Oh! This is lovely! This definitely fills a void -- the old MNI-Poly-AMU template page had nice-looking images too, but I hadn't made replacements for PAM50. So, thank you for taking the time to do this!

**Comment by joshuacwnewton** [comment]
This looks fantastic as part of the webpage. Awesome addition, @jcohenadad! :tada: 

https://spinalcordtoolbox.com/en/jn-2829-transfer-sourceforge-docs/overview/concepts/pam50.html

File: documentation/source/overview/concepts/pam50.rst
Comment (jcohenadad) [inline documentation/source/overview/concepts/pam50.rst:77]:
```suggestion
* `Levy S, Benhamou M, Naaman C, Rainville P, Callot V, Cohen-Adad J. White matter atlas of the human spinal cord with estimation of partial volume effect. Neuroimage, 2015, 119:262-271.<https://paperpile.com/shared/fK02OV>`_.
```

File: documentation/source/overview/concepts/pam50.rst
Comment (jcohenadad) [inline documentation/source/overview/concepts/pam50.rst:76]:
```suggestion
* `De Leener B, Fonov VS, Collins DL, Callot V, Stikov N, Cohen-Adad J. PAM50: Unbiased multimodal template of the brainstem and spinal cord aligned with the ICBM152 space. Neuroimage, 2018, 165:170-179. <https://paperpile.com/app/p/e74eced0-7d51-08e2-9c18-a34328fb4a86>`_.
```

File: documentation/source/overview/concepts/pam50.rst
Comment (jcohenadad) [inline documentation/source/overview/concepts/pam50.rst:78]:
```suggestion
* `Cadotte DW, Cadotte A, Cohen-Adad J, Fleet D, Livne M, Wilson JR, Mikulis D, Nugaeva N, Fehlings MG. Characterizing the location of spinal and vertebral levels in the human cervical spinal cord. AJNR Am J Neuroradiol, 2015, 36(4):803-810.<https://paperpile.com/app/p/5b580317-6921-06c8-a2ee-685d4dbaa44c>`_.
```

**Comment by jcohenadad** [comment]
@joshuacwnewton can we merge this? thank you for this titanesque work

**Comment by joshuacwnewton** [comment]
@jcohenadad Ah, my apologies, I had been awaiting another review. (There had been changes since your approval was originally given.)

If you're good with it, then I am too. Feel free to merge. :)

**Pull request merged**
